sale_commission_settlement_payment_date/wizard/wizard_settle.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Nextev
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import json
+from datetime import datetime
+
+from odoo import fields, models
+
+
+class SaleCommissionMakeSettle(models.TransientModel):
+    _inherit = "sale.commission.make.settle"
+
+    date_payment_to = fields.Date("Date Payment Up to", default=fields.Date.today())
+
+    def _get_agent_lines(self, agent, date_to_agent):
+        """
+        Exclude invoices with invoice payment date is after date_payment_to
+        """
+        agent_lines = super()._get_agent_lines(agent, date_to_agent)
+        for line in agent_lines.filtered(
+            lambda r: r.commission_id.invoice_state == "paid"
+            and self.date_payment_to != "false"
+        ):
+            if (
+                line.invoice_id.invoice_payments_widget != "false"
```

Comment:
Same as above. It's a text field. It'll probably be empty (`""` An empty string) or `False`, but never the string `"false"`. Might as well just do an `if line.invoice_id.invoice_payments_widget`

Reply:
@aleuffre Thanks for your review. I know that sound weird, but invoice_payments_widget it is possible to be a string "false".

Reply:
It happens when an invoice is not paid yet. I could change condition on invoice.state without checking invoice_payments_widget. What could you recommend me?